connector/connect/common/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -95,13 +96,49 @@ message Relation {
     // Catalog API (experimental / unstable)
     Catalog catalog = 200;
 
+    // ML relation
+    MlRelation ml_relation = 300;
+
     // This field is used to mark extensions to the protocol. When plugins generate arbitrary
     // relations they can add them here. During the planning the correct resolution is done.
     google.protobuf.Any extension = 998;
     Unknown unknown = 999;
   }
 }
 
+message MlRelation {
+  oneof ml_relation_type {
+    ModelTransform model_transform = 1;
+    FeatureTransform feature_transform = 2;
+    ModelAttr model_attr = 3;
+    ModelSummaryAttr model_summary_attr = 4;
+  }
+  message ModelTransform {
+    Relation input = 1;
+    ModelRef model_ref = 2;
+    MlParams params = 3;
+  }
+  message FeatureTransform {
+    Relation input = 1;
+    MlStage transformer = 2;
+  }
+  message ModelAttr {
```

Comment:
Any example of model attribute that returns Dataframe?

Reply:
model.summary.roc